ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt

```diff
@@ -143,26 +147,44 @@ class HomeScreenFragment :
       R.id.sync_status -> homeScreenViewModel.showSyncStatus()
       R.id.nav_offline_areas -> homeScreenViewModel.showOfflineAreas()
       R.id.nav_settings -> homeScreenViewModel.showSettings()
-      R.id.nav_sign_out -> showSignOutConfirmationDialog()
       R.id.about -> homeScreenViewModel.showAbout()
       R.id.terms_of_service -> homeScreenViewModel.showTermsOfService()
     }
     closeDrawer()
     return true
   }
 
-  private fun showSignOutConfirmationDialog() {
+  private fun showSignOutConfirmationDialogs() {
     // Note: Adding a compose view to the fragment's view dynamically causes the navigation click to
     // stop working after 1st time. Revisit this once the navigation drawer is also generated using
     // compose.
     binding.composeView.apply {
       setContent {
-        val openDialog = remember { mutableStateOf(true) }
-
-        // Reset the state for recomposition
-        openDialog.value = true
-
-        AppTheme { SignOutConfirmationDialog(openDialog) { userRepository.signOut() } }
+        val showUserDetailsDialog = remember { mutableStateOf(true) }
+        val showSignOutDialog = remember { mutableStateOf(false) }
+
+        // reset the state for recomposition
+        showUserDetailsDialog.value = true
+        showSignOutDialog.value = false
+
+        val user = homeScreenViewModel.userDetails.value!!
```

Comment:
This fragment already injects `UserRepository`. So, we don't need to fetch the details from view model. Instead, we can do something like `userRepository.getAuthenticatedUser()` directly here.

Reply:
Thank you for the suggestion @shobhitagarwal1612! I have changed it now.